assets/js/editor-components/utils/index.js

```diff
@@ -25,6 +25,7 @@ const getProductsRequests = ( {
 	const defaultArgs = {
 		per_page: isLargeCatalog ? 100 : 0,
 		catalog_visibility: 'any',
+		stock_status: [ 'instock', 'outofstock', 'onbackorder' ],
```

Comment:
This nicely pulls in out of stock items 👌.

But it doesn't exclude hidden items - now the question is do we want to show hidden items in spirit of what I mentioned where user should be able to design pages for future use - Or should we protect user from publishing things that are hidden? 

We do not have a system that would prevent such page/blocks to be published.

Also it seems if we create a block with a product and then set its visibility to `Hidden` the block with product will still render normally and won't be hidden on a published page. 

As a user I'd be confused that I marked something hidden and it's still showing up.

So in general this is an improvement I think question is if we want to tackle hidden as part of this or separate issue? I feel like handling hidden products might be a bigger thing!

Reply:
Good question. Let's tackle the hidden product's issue separately, I'll follow up in a P2. 👍